### PR TITLE
[INFINITY-23222] Fix bootstrap networking shim - backport

### PR DESCRIPTION
**Pull request by benclarkwood**
1. Fix logic in container IP lookup. (#1515)
2. Enhance logging in container IP lookup and assignment.

### CODE REVIEW

**Comment by elezar** [comment]
This is the backport of #1515, right, so this isn't the place to make style comments.

File: sdk/bootstrap/ip.go [hunk 1]
```diff
@@ -6,38 +6,42 @@ import (
...
+	validIP := net.ParseIP(ip)
+	if validIP == nil {
+		return nil, fmt.Errorf("Invalid %s found: %s", envvar, ip)
 	}
 
-	ipAddr, err := net.ResolveIPAddr("ip", hostName)
-	if err != nil {
-		err = fmt.Errorf("Unable to resolve hostname(%s): %s", hostName, err)
-		return
+	if validIP.String() == "0.0.0.0" || validIP.String() == "::" {
```
Comment (elezar) [inline sdk/bootstrap/ip.go]: This check here implies support for IPv6 addresses, but the tests cases below don't exercise this.

File: sdk/bootstrap/main.go [hunk 2]
```diff
@@ -336,52 +336,43 @@ func isFile(path string) (bool, error) {
...
 
-	pod_ip, err := GetLocalIP()
+	log.Printf("Looking up container IP address...")
+	containerIP, err := getContainerIPAddress()
 	if err != nil {
-		log.Fatalf("Cannot find the container's IP address: ", err)
+		log.Fatalf("Cannot look up container's IP address: %s", err)
+	}
+	log.Printf("Container IP address was found to be: %s", containerIP)
+	// Set both LIBPROCESS_IP and MESOS_CONTAINER_IP so that SDK developers
```
Comment (elezar) [inline sdk/bootstrap/main.go]: Thise could also be done in a loop like @asridharan suggested in #1515.

File: sdk/bootstrap/ip.go (hunk 1 above)
Comment (benclarkwood) [reply sdk/bootstrap/ip.go]: That is true. I was lazy.

File: sdk/bootstrap/main.go (hunk 2 above)
Comment (benclarkwood) [reply sdk/bootstrap/main.go]: I personally don't like the loop style, as someone can easily swap the order of an array.

Comment (benclarkwood) [reply sdk/bootstrap/main.go]: (it also IMO sacrifices readability for saving a handful of lines)

**Pull request merged**